src/st/cat/e_bone_ark.c

```diff
@@ -0,0 +1,1466 @@
+                        prim->next->b3 = 0x80;
+                        prim->next->tpage = self->rotate;
+                        prim->next->r3 = i;
+                        LOH(prim->next->r2) = 0x28;
+                        LOH(prim->next->b2) = 0xC;
+                        prim->priority = self->zPriority;
+                        prim->drawMode = DRAW_UNK02;
+                        // nb. this looks like a copy paste mistake
+                        // this is overwritten by the loop iteration
+                        prim = prim->next;
+                    }
+                }
+            } else {
+                self->ext.boneArk.prim = NULL;
+            }
+
+            for (i = 1; i < 10; i++) {
+                entity = &self[i];
+                DestroyEntity(entity);
+            }
+
+            self->animCurFrame = 0;
+            self->step_s++;
+            break;
+        case 2:
+            var_s5 = 0;
+            prim = self->ext.boneArk.prim;
+            while (prim != NULL) {
+                if (prim->p3 & 8) {
+                    var_s5 |= 1;
+                    RenderDeathParts(prim);
+                }
+                prim = prim->next;
+            }
+
+            if (!var_s5) {
+                PreventEntityFromRespawning(self);
+                DestroyEntity(self);
+                return;
+            }
+
+            break;
+        }
+        break;
+    case 16:
+#include "../pad2_anim_debug.h"
+    }
+
+    xOffset = (((self - 1)->ext.boneArk.skeletonPosX +
+                (self - 2)->ext.boneArk.skeletonPosX) /
+               2) +
+              3;
+    yOffset = ((self - 1)->ext.boneArk.skeletonPosY +
+               (self - 2)->ext.boneArk.skeletonPosY) /
+              2;
+    self->posX.i.hi = xOffset;
+    self->posY.i.hi = yOffset;
+    dx = (self - 2)->ext.boneArk.skeletonPosX -
+         (self - 1)->ext.boneArk.skeletonPosX;
+    dy = (self - 2)->ext.boneArk.skeletonPosY -
+         (self - 1)->ext.boneArk.skeletonPosY;
+    angle = ratan2(dy, dx);
+    self->rotate = angle;
+
+    var_s3 = self->ext.boneArk.unk88.i.lo;
+    angle = -self->rotate;
+    angle = (var_s3 - angle) / 4;
+    for (i = 5; i < 7; i++) {
+        entity = &self[i];
+        var_s3 = var_s3 - angle;
+        entity->ext.boneArk.unk80 = var_s3 + self->rotate;
+    }
+
+    var_s3 = -self->rotate;
+    angle = ((self + 9)->ext.boneArk.unk80 - var_s3) / 3;
+    for (i = 7; i < 9; i++) {
+        entity = &self[i];
+        var_s3 = var_s3 + angle;
+        entity->ext.boneArk.unk80 = var_s3 + self->rotate;
+    }
+}
+
+// nb. This must be defined after pad2_anim_debug.h to align PSP data
+static s16 skeleton_yOffsets[] = {
+    10, -20, -21, -22, -8, -20, -19, -8, -6, 0, 4, 8, 10, 0, 0, 0};
+
+void EntityBoneArkSkeleton(Entity* self) {
+    Entity* otherSkeletonEntity;
+    Entity* boneArkEntity;
+    Entity* newEntity;
+    u8 collisionDetected;
+
+    typedef enum Step {
```

Comment:
Kind of just randomly commenting here, but it seems like the placement of the enums is a bit inconsistent.  This one is in the function, but a previous one was in between functions.

Also, I see these being typedef a lot.  I don't know that it matters one way or the other, but being a typedef is only needed if you're going to use it as a type and not if you're just using the elements.

Reply:
The one used by multiple entities is at file level scope, the other is only used within a single entity so I put it at function level scope.

I know this has been discussed briefly before but I don't know if a conclusion was reached, happy to change this up for consistency with other places if required.